slack_bot/slash_commands/api.py

```diff
@@ -0,0 +1,112 @@
+    more_than_thirty_seconds = now - x_slack_request_timestamp > 30
+    if more_than_thirty_seconds:
+        raise HTTPException(400, "Invalid timestamp")
+    return x_slack_request_timestamp
+
+
+async def with_body(request: Request) -> bytes:
+    """Return request body.
+
+    Per design, a route cannot depend on a form field and consume body because
+    dependencies resolution consumes body.
+    Therefore, this is not using `fastapi.Form` to extract form data parameters.
+    """
+    return await request.body()
+
+
+def with_form_data(body: bytes = Depends(with_body)) -> dict:
+    return dict(parse_qsl(body.decode()))
+
+
+def check_signature(secret: str, timestamp: int, signature: str, body: bytes) -> bool:
+    """Return True if signature is valid and False if not."""
+    signature_message = f"v0:{timestamp}:{body.decode()}"
+    local_hash = HMAC(
+        secret.encode(), signature_message.encode(), sha256).hexdigest()
+    local_signature = f"v0={local_hash}"
+    return local_signature == signature
+
+
+def with_settings() -> Settings:
+    try:
+        return Settings()
+    except ValidationError as error:
+        raise HTTPException(500) from error
+
+
+def with_valid_signature(
+    body: bytes = Depends(with_body),
+    settings: Settings = Depends(with_settings),
+    timestamp: int = Depends(with_valid_timestamp),
+    signature: str = Header(..., alias="X-Slack-Signature"),
+) -> str:
+    secret = settings.signing_secret.get_secret_value()
+    if check_signature(secret, timestamp, signature, body) is False:
+        raise HTTPException(403, "invalid signature")
+    return signature
+
+
+class Command(BaseModel):
+    token: str
+    team_id: str
+    team_domain: str
+    channel_id: str
+    channel_name: str
+    user_id: str
+    user_name: str
+    command: str
+    response_url: str
+    trigger_id: str
+    api_app_id: str
+    text: str = None
+    enterprise_id: str = None
+    enterprise_name: str = None
+
+    def __init__(
+        self,
+        form_data: dict = Depends(with_form_data),
+        signature=Depends(with_valid_signature),
+    ):
+        super().__init__(**form_data)
+
+
+@router.post('/process_command')
+def process_command(command: Command = Depends()):
+    text = get_answear(command.command[1:])
+    try:
+        client.chat_postMessage(channel=command.channel_id,
+                        text="",
+                        unfurl_links=False,
+                        blocks=[
+                            {
+                                "type": "section",
+                                "text": {
+                                    "type": "mrkdwn",
+                                    "text": text
+                                }
+                            }
+                        ],
+                    )
+    except Exception as e:
+        logger.error(f"Error in processing command '{command.command}' : ", exc_info=e)
+        return Response(status_code=500)
```

Comment:
Why hardcoded to 500? What if 403 or any other 400th response is returned?

Reply:
I think the status code from the server should be parsed and returned to the user as it was received from the server

Reply:
Doesn't matter what was retuned. Slack will show that the bot is not working if it will not response 200 
<img width="638" alt="Знімок екрана 2023-08-20 о 15 54 22" src="https://github.com/COXIT-CO/CVScanAI/assets/47090110/64f43096-fc49-46db-8d77-169f29b075b8">

Reply:
so if I want to debug it how should I know why exactly slack failed? Error handling its not about what users see, its about what developers see int he log while users reports bot not working.

Reply:
the response after the logging will not change anything

Reply:
It will not, but it should. I would add log message before the exception code (since not-success response will not throw exception I guess) - if status code is not 200 then print out warning.

The case I want to cover here is what if I provide wrong Slack API key, how would I be notified so I can fix it quickly.

Reply:
lets fix it in #14